core/opendaq/modulemanager/src/module_manager_impl.cpp

```diff
@@ -1038,6 +1060,22 @@ ServerCapabilityPtr ModuleManagerImpl::mergeDiscoveryAndDeviceCap(const ServerCa
     return merged.detach();
 }
 
+void ModuleManagerImpl::copyGeneralProperties(const PropertyObjectPtr& general, const PropertyObjectPtr& tartgetObj)
+{
+    if (!general.assigned())
+        return;
+
+    for (const auto& prop : general.getAllProperties())
+    {
+        const auto name = prop.getName();
+        const auto value = general.getPropertyValue(name);
+        const auto defaultValue = prop.getDefaultValue();
+
+        if (tartgetObj.hasProperty(name) && tartgetObj.getPropertyValue(name) == defaultValue)
+            tartgetObj.setPropertyValue(name, value);
+    }
+}
```

Comment:
Can't we just use clone here?

Reply:
Unfortunately no. I want the property to be copied from general object to target object only if the value is not present in the target object. This means that if the user sets username and password on the general config object and on the device config object, the values from the more specific device config object will be used.